config/api/routes/lock.php

```diff
@@ -9,7 +8,13 @@
     [
         'pattern' => '(:all)/lock',
         'method'  => 'GET',
+        /**
+         * @deprecated 3.6.0
+         * @todo Remove in 3.7.0
+         */
         'action'  => function (string $path) {
+            deprecated('The `GET (:all)/lock` API endpoint has been deprecated and will be removed in 3.7.0');
```

Comment:
Why the deprecation of the lock/unlock getters? These API routes might still make sense for custom applications and integrations.

Reply:
Our current content locking isn't great - I think we need to move rather soon a version based on storing unsaved changes not he server etc. - That will likely change a lot, which is why I would really like to discourage the use of these endpoints already by removing them.

And let's face it, any discussion here is pretty sure a theoretical one - we've never documented this properly, I don't think the content locking API has been useable at all for developers so far.

Reply:
Makes sense 👍